docs/ref/forms/api.txt

```diff
@@ -552,15 +592,32 @@ The template used to render a field's ``<label>``, used when calling
 form by overriding this attribute or more generally by overriding the default
 template, see also :ref:`overriding-built-in-form-templates`.
 
+Output styles
+-------------
+
+As well as rendering the form with ``print(form)``, or in a template
+with ``{{ form }}``, the following helper functions serve as a proxy to
+:meth:`Form.render` passing a particular ``template_name`` value.
+
+These helpers are most useful in a template, where you need to override the
+form renderer or form provided value but cannot pass the additional parameters
+to :meth:`~Form.render`. For example, you can render a form with as an
+unordered list using ``{{ form.as_ul }}``.
+
+Each helper pairs a form method with an attribute giving the appropriate
+template name.
+
```

Comment:
_Maybe_ we could add a note as part of this change to say the current methods are not recommended, as suggested in [comment](https://code.djangoproject.com/ticket/32339#comment:4).

Here's an initial draft. Likely this is too verbose. 

```
.. note::

    The ``as_p()``, ``as_ul()`` and ``as_table()`` methods use templates which 
    are suitable for rendering a form quickly. However, they do not implement 
    accessibility best practices, such as grouping related ``<inputs>`` in a 
    ``<fieldset>`` with a ``<legend>``. Furthermore ``as_table()`` can result 
    in forms which can be confusing for screen reader users. These methods are 
    therefore not recommended for production code. 
```

Reply:
Hey @smithdc1 — good point yes... 

I rephrased as: 

```

.. note::

    The ``as_p()``, ``as_ul()`` and ``as_table()`` methods are suitable for
    rendering a form quickly. However, the default templates they use do not
    implement accessibility best practices, such as grouping related
    ``<inputs>`` in a ``<fieldset>`` with a ``<legend>``. Furthermore
    ``as_table()`` can result in forms which can be confusing for screen reader
    users. Therefore, these methods are not recommended for production code.
```

I think though we should add it when adding the `div` template — otherwise we've got a warning with no workaround... (So let's pull it in there momentarily. 😄 )